packages/tools/playground/src/tools/monacoManager.ts

```diff
@@ -220,126 +210,30 @@ class Playground {
-        const useAnimation = Utilities.ReadStringFromStore("useAnimation", "None");
-        const useParticles = Utilities.ReadStringFromStore("useParticles", "None");
-        const useSnippet = Utilities.ReadStringFromStore("useSnippet", "None");
-        const useImport = Utilities.ReadStringFromStore("useImport", "None");
-
-        // Build code
-        let code = "";
-
-        // Scene
-        code += "\n// Create a new Scene :\n";
-        code += "var scene = new BABYLON.Scene(engine);\n";
-
-        if (debugLayer) {
-            code += "\n// Show the inspector :\n";
-            code += this._getCode("debugLayer");
-        }
-
-        // Camera
-        code += "\n// Create a camera and attach it to canvas :\n";
-        code += this._getCode(useCamera);
-
-        // Light
-        code += "\n// Create light :\n";
-        code += this._getCode(useLight);
-
-        // Shadows
-        if (useShadows) {
-            code += "\n// Create a shadow generator :\n";
-            if (useLight == "Hemispheric") {
-                code += "// Shadows cannot be generated with Hemispheric light !\n";
-            } else {
-                code += this._getCode("useShadows");
-            }
-        }
-
-        // Meshes
-        if (useGround || useSphere || useBox || useCylinder) {
-            code += "\n// Add geometry\n";
-        }
-        if (useGround) {
-            code += this._getCode("useGround");
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code += "ground.receiveShadows = true;\n";
-            }
-        }
-        if (useSphere) {
-            code += this._getCode("useSphere");
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code += "shadowGenerator.addShadowCaster(sphere);\n";
-            }
-        }
-        if (useBox) {
-            code += this._getCode("useBox");
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code += "shadowGenerator.addShadowCaster(box);\n";
-            }
-        }
-        if (useCylinder) {
-            code += this._getCode("useCylinder");
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code += "shadowGenerator.addShadowCaster(cylinder);\n";
-            }
-        }
-
-        // Animations
-        if (useAnimation != "None") {
-            code += "\n// Import animated character :\n";
-            code += this._getCode(useAnimation);
-            if (this._templates.length && useShadows && useLight != "Hemispheric") {
-                code = code.slice(0, code.length - 4) + "    shadowGenerator.addShadowCaster(hero);\n});\n";
+    private _insertCodeAtCursor(code: string) {
+        if (this._editor) {
+            // Get the current position of the cursor
+            const position = this._editor?.getPosition();

```

Comment:
nit - `this._editor` is already defined, no need for `?`

Reply:
Indeed :+1: Last commit is the fix